beacon/dkg.go

```diff
@@ -376,7 +390,42 @@ func (dkg *DistributedKeyGeneration) checkMsg(msg *types.DKGMessage, index int,
 		return fmt.Errorf("checkMsg: incorrect dkgIteration %v", msg.DKGIteration)
 	}
 	if len(msg.ToAddress) != 0 && !bytes.Equal(msg.ToAddress, dkg.privValidator.GetPubKey().Address()) {
-		return fmt.Errorf("checkMsg: not ToAddress")
+		return fmt.Errorf("checkMsg: ToAddress not to us ")
+	}
+	if !val.PubKey.VerifyBytes(msg.SignBytes(dkg.chainID), msg.Signature) {
+		return fmt.Errorf("checkMsg: failed signature verification")
+	}
+	return nil
+}
+
+// Validate that the message is a valid one to be taking part in the DKG in general (not necessarily to us)
+func (dkg *DistributedKeyGeneration) validateMessage(msg *types.DKGMessage, index int, val *types.Validator) error {
```

Comment:
need this as the gatekeeper to the mempool - slightly different from checkMsg in that your own messages should pass through too (ALL txs are checked going into the mempool including your own)

Reply:
There is so much overlap between these functions if would be neater to refactor to just keep one.

Reply:
you're right. I will do this